File: mechanic_server/src/models/vehicle.ts
export {};
const {model, Schema} = require('mongoose');


const VehicleSchema = Schema({
	userId: Schema.Types.ObjectId,
	brand: {
		type: String
	},
	make: {
		type: String
	},
	model: {
		type: String
	},
	plateNumber: {
		type: String
	},
	color: {
		type: String
	}
});

module.exports = model('vehicle', VehicleSchema);

File: mechanic/lib/models/vehicle.dart
class Vehicle {
  Vehicle(
      {required this.brand,
      required this.make,
      required this.model,
      required this.plateNumber,
      required this.color,
      this.vehicleId = ""
      });

  final String brand, make, model, plateNumber, color, vehicleId;

  static Vehicle fromJson(Map<String, dynamic> data) {
    return Vehicle(
        brand: data['brand'],
        make: data['make'],
        color: data['color'],
        model: data['model'],
        plateNumber: data['plateNumber'],
        vehicleId: data["_id"]
        );
  }

  Map<String, dynamic> toJson() {
    return {
      "brand": brand,
      "make": make,
      "model": model,
      "plateNumber": plateNumber,
      "color": color
    };
  }
}
